src/topp/QualityControl.cpp

```diff
@@ -177,6 +247,51 @@ class TOPPQualityControl : public TOPPBase
     }
     return files;
   }
+
+  // templated function to copy all meta values from one object to another
+  template <class FROM, class TO>
+  //TODO get a MetaValue list to copy only those that have been set
+  void copyMetaValues_(FROM& from, TO& to)
+  {
+    vector<String> keys;
+    from.getKeys(keys);
+    for(String& key : keys)
+    {
+      to.setMetaValue(key, from.getMetaValue(key));
+    }
+  }
+
+  void copyPepIDMetaValues_(vector<PeptideIdentification>& pep_ids, const map<Int64,PeptideIdentification*>& map_to_id)
+  {
+    for (PeptideIdentification& pep_id : pep_ids)
+    {
+      if (!pep_id.metaValueExists("UID")) // PepID doesn't has ID, needs to have MetaValue
+      {
+        cerr << "No unique ID at unassigned peptideidentifications found. Please run PeptideIndexer with '-addUID'.\n";
+        exit(ILLEGAL_PARAMETERS);
+      }
+      PeptideIdentification ref_pep_id = *(map_to_id.at(pep_id.getMetaValue("UID"))); // look up annotated PepID
+
+      // copy all MetaValues that are at PepID level
+      copyMetaValues_(ref_pep_id,pep_id);
+
+      // copy all MetaValues that are at Hit level
+      copyMetaValues_(ref_pep_id.getHits()[0],pep_id.getHits()[0]);
+    }
+  }
+
+  void fillPepIDMap_(map<Int64,PeptideIdentification*>& map_to_id, vector<PeptideIdentification>& pep_ids)
```

Comment:
change to
`void fillPepIDMap_(map<Int64,PeptideIdentification*>& map_to_id, const vector<PeptideIdentification>& pep_ids) const`
(and add internal constness)